tests/network/localnet/liblocalnet.py

```diff
@@ -21,7 +21,8 @@
 
 def run_vms(vms: tuple[BaseVirtualMachine, ...]) -> tuple[BaseVirtualMachine, ...]:
     for vm in vms:
-        vm.start()  # type: ignore[no-untyped-call]
+        if not vm.vmi.exists:
```

Comment:
This is raceful and not recommended.
It will also change the function behavior from actually running VMs to accepting currently-running ones.

I have no special problem with the second, although it feels it will bite us on a different scenario.
I do have a problem with asking for permission by checking the VMI existance, which is also too low level thing. I suggest you wrap this with a try-except and explicitly ignore the error of not being able to start it on the specific reason that it is already running.